src/tests/ftest/util/host_utils.py

```diff
@@ -0,0 +1,338 @@
+    def clients_with_localhost(self):
+        """Get the test clients including the localhost.
+
+        Returns:
+            NodeSet: test clients including the localhost
+
+        """
+        return self.clients | NodeSet(get_local_host())
+
+    def display(self, log):
+        """Display the host information.
+
+        Args:
+            log (logger): logger for the messages produced by this method
+        """
+        log.info("-" * 100)
+        log.info("--- HOST INFORMATION ---")
+        log.info("hostlist_servers:    %s", self.servers)
+        log.info("hostlist_clients:    %s", self.clients)
+        log.info("server_partition:    %s", self._servers.partition.name)
+        log.info("client_partition:    %s", self._clients.partition.name)
+        log.info("server_reservation:  %s", self._servers.partition.reservation)
+        log.info("client_reservation:  %s", self._clients.partition.reservation)
+        log.info("access_points:       %s", self.access_points)
+
+    def set_hosts(self, log, server_hosts, server_partition, server_reservation, client_hosts,
+                  client_partition, client_reservation):
+        """Set the hosts.
+
+        Args:
+            log (logger): logger for the messages produced by this method
+            server_hosts (object): hosts to define as servers
+            server_partition (str): server partition name
+            server_reservation (str): server reservation name
+            client_hosts (object): hosts to define as clients
+            client_partition (str): client partition name
+            client_reservation (str): client reservation name
+
+        Raises:
+            HostException: if there is a problem obtaining the hosts
+
+        """
+        self._servers = HostRole(server_hosts, server_partition, server_reservation)
+        self._servers.update_hosts(log)
+        self._clients = HostRole(client_hosts, client_partition, client_reservation)
+        self._clients.update_hosts(log)
+
+        # Optionally remove any servers that may have ended up in the client list.  This can occur
+        # with tests using slurm partitions as the client partition is setup with all of the hosts.
+        self._update_clients(log)
+
+    def set_test_hosts(self, test):
+        """Get the server and client hosts for this test.
+
+        Args:
+            test (Test): the avocado test class
+
+        Raises:
+            HostException: if there is a problem obtaining the hosts
+
+        """
+        server_params = get_host_parameters(
+            test, "test_servers", "server_partition", "server_reservation", "/run/hosts/*")
+        client_params = get_host_parameters(
+            test, "test_clients", "client_partition", "client_reservation", "/run/hosts/*")
+        self.set_hosts(test.log, *server_params, *client_params)
+
+    def _update_clients(self, log):
+        """Remove any servers from the set of clients if the clients were derived from a partition.
```

Comment:
Maybe to be more clear about what this does?
```suggestion
    def _remove_servers_from_clients(self, log):
        """Remove any servers from the set of clients if the clients were derived from a partition.
```

Reply:
No longer being used.